### ISSUE TITLE
Sprout Report - BUG/Issue Trying to get property 'name' of non-object

### ISSUE THREAD

**Issue by boldenamsterdam** [barrelstrength/craft-sprout-reports#72]
### Description (screenshots, logs, ...)
Hi Ben,

We are getting a PHP notice 'Trying to get property 'name' of non-object' on the staging environment.
Error's are being displayed on the **admin/sprout-reports/reports** and **admin/sprout-reports/datasources** pages.

![Screenshot 2019-12-24 at 11 58 33](https://user-images.githubusercontent.com/7466604/71410343-2e52d200-2645-11ea-825d-3966856c0206.png)

web.log
```[-][1][f015cc72e588a957bc513401d6683b7c][error][yii\base\ErrorException:8] yii\base\ErrorException: Trying to get property 'name' of non-object in /data/sites/web/websitename/subsites/staging.websitename.pro/shared/vendor/barrelstrength/sprout-base-reports/src/elements/Report.php:273```

We are not facing the issue on local dev machines.

### Additional info

It looks like it has something to do with the initialization of the plugin.
I got the plugin working after I made a change in the ```getPlugin()``` function.

```Changed line 67 of the file: sprout-base-reports/src/base/DataSource.php```

From:
```
final public function getPlugin()
    {
        return $this->plugin
    }
```

To:
```
final public function getPlugin()
    {
        return $this->plugin = Craft::$app->getPlugins()->getPlugin('sprout-reports');
    }
```

- Craft version: 3.2.9
- PHP version: 7.2
- Database driver & version:  
- Plugins & versions:

1. Commerce Currency Prices | 2.6.1
2. Contact Form | 2.2.5
3. Contact Form Honeypot | 1.0.2
4. Cookies | 1.1.11
5. Craft Commerce | 2.1.11
6. Element API | 2.6.0
7. GeoMate | v1.0.6
8. IMGix | 2.0.4
9. Instagram Feed 1.0.2
10. Inventory 2.0.3
11. MobileDetect 1.0.2
12. Mollie for Craft Commerce 2.1.0.1
13. Patrol  3.1.3
14. Redactor 2.3.3.2
15. Retour | 3.1.22
16. SEOmatic | 3.2.27
17. Sprout Report | 1.2.7
18. Twigpack | 1.1.8
19. Video Embedder | 1.1.3

If more info is needed please let us know.

**Comment by BenParizek** [comment]
It looks like we've had few small bugfix releases beyond your currently installed version of Sprout Reports where I believe we addressed this issue. Can you update to the latest (v1.2.11) and let me know if you continue to receive the error?

**Comment by BenParizek** [comment]
@boldenamsterdam 

As I haven't heard back on this I'm going to close it out. We just released an update with Craft 3.4 support. If you're still experiencing any issues in the latest please re-open or post a new ticket and we'll look into it more.

**Comment by JPBarringer** [comment]
Ran into this same problem even after updates -- solution was to delete `/vendor` and `composer install` to rebuild the dependency tree.
